core/src/main/java/com/contentful/vault/SyncConfig.java

```diff
@@ -104,6 +117,13 @@ public Builder setClient(CDAClient client) {
         );
       }
 
+      if (environment != null) {
+        throw new IllegalStateException(
+            "Do not set a client, when an environment is already set. User either environment " +
```

Comment:
@miron4dev same here. Also if we are throwing the same message I think we can extract the text to a constant, what do you think?

Reply:
Good idea. I'll do it tomorrow.